File: 11-ContainerWithMostWater/11-ContainerWithMostWater.c
// Last updated: 8/1/2025, 10:42:56 PM
int maxArea(int* height, int heightSize) {
    int left=0;
    int right=heightSize-1;
    int maxarea=0;
    while(left<right){
        int h=height[left]<height[right]?height[left]:height[right];
        int w=right-left;
        int area=h*w;
        if(area>maxarea){
            maxarea=area;

        }
        if(height[left]<height[right]){
            left++;
        }
        else{
            right--;
        }
    }
    return maxarea;
}

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 8/1/2025, 10:42:49 PM
class Solution {
    public int maxArea(int[] height) {
       int n=height.length;
       int i=0;
       int j=n-1;
       int maxWater=Integer.MIN_VALUE;
        while(i<j){
            int h=Math.min(height[i],height[j]);
            int w=j-i;
            maxWater=Math.max(maxWater,h*w);
            if(height[i]<height[j]){
                i++;
            }else{
                j--;
            }
        }
        return maxWater;
    }
}